examples/java/src/main/java/glide/examples/GlideFtExample.java

```diff
@@ -0,0 +1,228 @@
+        // Check `GlideClusterClientConfiguration` for additional options.
+        GlideClusterClientConfiguration config =
+                GlideClusterClientConfiguration.builder()
+                        .addresses(nodeList)
+                        // Enable this field if the servers are configured with TLS.
+                        // .useTLS(true);
+                        .build();
+
+        GlideClusterClient client = GlideClusterClient.createClient(config).get();
+        return client;
+    }
+
+    /**
+     * Executes the main logic of the application, performing basic operations such as FT.CREATE and
+     * FT.SEARCH using the provided <code>GlideClusterClient</code>.
+     *
+     * @param client An instance of <code>GlideClusterClient</code>.
+     * @throws ExecutionException if an execution error occurs during operations.
+     * @throws InterruptedException if the operation is interrupted.
+     */
+    public static void appLogic(GlideClusterClient client)
+            throws ExecutionException, InterruptedException {
+
+        String prefix = "{" + UUID.randomUUID() + "}:";
+        String index = prefix + "index";
+
+        CompletableFuture<String> createResponse =
+                FT.create(
+                        client,
+                        index,
+                        new FieldInfo[] {
+                            new FieldInfo("vec", "VEC", VectorFieldHnsw.builder(DistanceMetric.L2, 2).build())
+                        },
+                        FTCreateOptions.builder()
+                                .dataType(DataType.HASH)
+                                .prefixes(new String[] {prefix})
+                                .build()); // "OK"
+
+        CompletableFuture<Long> hsetResponse =
+                client.hset(
+                        gs(prefix + 0),
+                        Map.of(
+                                gs("vec"),
+                                gs(
+                                        new byte[] {
+                                            (byte) 0, (byte) 0, (byte) 0, (byte) 0, (byte) 0, (byte) 0, (byte) 0, (byte) 0
+                                        }))); // response is 1L which represents the number of fields that were added.
+
+        hsetResponse =
+                client.hset(
+                        gs(prefix + 1),
+                        Map.of(
+                                gs("vec"),
+                                gs(
+                                        new byte[] {
+                                            (byte) 0,
+                                            (byte) 0,
+                                            (byte) 0,
+                                            (byte) 0,
+                                            (byte) 0,
+                                            (byte) 0,
+                                            (byte) 0x80,
+                                            (byte) 0xBF
```

Comment:
Is this necessary? I think by saying this is a `new byte[]`, the type of the elements is known statically, which allows us to avoid all the casting.

Reply:
applies throughout